persistent/test/main.hs

```diff
@@ -22,7 +22,18 @@ import Data.Time
 import Text.Shakespeare.Text
 
 import Database.Persist.Class.PersistField
+import Database.Persist.Quasi
 import Database.Persist.Quasi.Internal
+       ( Line(..)
+       , LinesWithComments(..)
+       , Token(..)
+       , associateLines
+       , parseFieldType
+       , parseLine
+       , preparse
+       , splitExtras
+       , takeColsEx
+       )
```

Comment:
Likewise with this test, we are importing `Quasi.Internal`, but I assume, long term, we want to test the main module really without touching the `Internal` module (except other than perhaps the internal types, `Line`/`Token` and so on). I can revert this if you want though.

Having worked a little now on these modules/tests, I think it might be nice (especially after this `Internal` split on the quasi module), to have a test file for `Quasi` and a test file for `Quasi.Internal`, though I feel like a lot of these tests in here were added initially to cover as much of the module as possible because of the initial lack of coverage, it might be that a test file for `Quasi` to test the expected parsing behaviour would be sufficient. 

Just a thought I had while making this change :+1:

Reply:
Yeah I'd eventually like to hollow this out into a more 'typical' `hspec` test suite, so we'd have `Database.Persist.QuasiSpec` etc that contains all this stuff.